src/app/onion/learning-object-builder/builder-store.service.ts

```diff
@@ -449,20 +450,23 @@ export class BuilderStore {
 
   /**
    * Updates Url at given index
-   *
+   * Also checks for valid Url and title field input against the supplied regex pattern
    * @param {number} index
    * @param {Url} url
    * @memberof BuilderStore
    */
   private async updateUrl(index: number, url: Url): Promise<any> {
-    if (!url.url.match(/https?:\/\/.+/i)) {
-      url.url = `http://${url.url}`;
-    }
-    this.learningObject.materials.urls[index] = url;
-    await this.saveObject({
-      'materials.urls': this.learningObject.materials.urls
-    });
-    this.learningObjectEvent.next(this.learningObject);
+    var exp = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/;
+    if (url.url.match(exp) && url.title !== '') {
+      this.learningObject.materials.urls[index] = url;
+      await this.saveObject({
+        'materials.urls': this.learningObject.materials.urls
+      });
+      this.learningObjectEvent.next(this.learningObject);
+    } else {
```

Comment:
What's the reason for returning null here, as compared to the function naturally returning void?

Reply:
You make a good point, I originally added it because naively I thought we needed to return something but now I see that we only want this function to run only if it true